packages/exo/src/exo-tools.js

```diff
@@ -235,6 +235,7 @@ export const defendPrototype = (
       methodGuards && methodGuards[prop],
     );
   }
+
```

Comment:
This doesn't look right. When I compare endo and agoric-sdk, I get substantive diffs in this file:

```
~/projects/endo
17:21 connolly@bldbox$ diff -u packages/exo/src/exo-tools.js ~/projects/agoric-sdk/packages/store/src/patterns/exo-tools.js 
--- packages/exo/src/exo-tools.js       2023-04-03 17:19:54.297144138 -0500
+++ /home/connolly/projects/agoric-sdk/packages/store/src/patterns/exo-tools.js 2023-04-01 14:03:32.978861672 -0500
@@ -1,18 +1,12 @@
 import { E, Far } from '@endo/far';
-import { listDifference, objectMap, mustMatch, M } from '@endo/patterns';
+import { listDifference, objectMap } from '@agoric/internal';
+
+import { MinMethodGuard, mustMatch, M } from './patternMatchers.js';
 
 const { quote: q, Fail } = assert;
 const { apply, ownKeys } = Reflect;
 const { defineProperties } = Object;
 
-/**
- * A method guard, for inclusion in an interface guard, that enforces only that
- * all arguments are passable and that the result is passable. (In far classes,
- * "any" means any *passable*.) This is the least possible enforcement for a
- * method guard, and is implied by all other method guards.
- */
-const MinMethodGuard = M.call().rest(M.any()).returns(M.any());
-
 const defendSyncArgs = (args, methodGuard, label) => {
   const { argGuards, optionalArgGuards, restArgGuard } = methodGuard;
   const paramsPattern = M.splitArray(
@@ -235,6 +229,7 @@
       methodGuards && methodGuards[prop],
     );
   }
+
   return Far(tag, /** @type {T} */ (prototype));
 };
 harden(defendPrototype);
```

Reply:
In agoric-sdk, `MinMethodGuard` is created and exported by `patternMatchers.js` and imported into `exo-tools.js`.

In endo, `MinMethodGuard` is still created and exported by `patternMatchers.js`, and it is still needed by `exo-tools.js`. However, now these are in separate packages. `patternMatchers.js` is in the `@endo/patterns` package, which does not export `MinMethodGuard`, so it would be harder for `exo-tools.js` to import it from there.

Repairs could be
   * revise `@endo/patterns` to directly export `MinMethodGuard`. But it seems to specialized to treat it as a general export of the package
   * revive `@endo/patterns` so that it is separately exported, via an `"exports":` in the package.json.
   * just redundantly define it is `exo-tools.js`, which is the *only* place it is needed.

I elected to do the last. But, with your question provoking me to discover that *only* exo-tools.js uses it, I am about to remove its redundant definition from the endo `patternMatchers.js`.

To keep the repositories in sync, I will also create a separate PR for agoric-sdk that moves `MinMethodGuard` from `patternMatchers.js` to `exo-tools.js`.

Reply:
Done here, and opened https://github.com/Agoric/agoric-sdk/pull/7320 . Resolving.